packages/selected-network-controller/src/SelectedNetworkController.ts

```diff
@@ -236,6 +236,24 @@ export class SelectedNetworkController extends BaseController<
     });
   }
 
+  #unsetNetworkClientIdForDomain(domain: Domain) {
+    const globallySelectedNetworkClient = this.messagingSystem.call(
+      'NetworkController:getSelectedNetworkClient',
+    );
+    const networkProxy = this.#proxies.get(domain);
+    if (networkProxy && globallySelectedNetworkClient) {
+      networkProxy.provider.setTarget(globallySelectedNetworkClient.provider);
+      networkProxy.blockTracker.setTarget(
+        globallySelectedNetworkClient.blockTracker,
+      );
+    } else if (networkProxy) {
+      this.#proxies.delete(domain);
```

Comment:
we still need to destroy the proxies to avoid a memory leak. 

From @Gudahtt :

We can counteract this by storing the proxies as WeakRefs using the strategy described here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Memory_management#weakrefs_and_finalizationregistry